docs/development/contributing.md

```diff
@@ -8,17 +8,56 @@ for diving into it.
 
 ## Development Workflow
 
-See {ref}`building_from_sources` and {ref}`testing` documentation.
+To contribute code, see the following steps,
 
-For code-style the use of [pre-commit](https://pre-commit.com/) is also recommended,
+1. Fork the Pyodide repository [https://github.com/pyodide/pyodide](https://github.com/pyodide/pyodide) on Github.
+2. If you are on Linux, you can skip this step. On Windows and MacOS you have a
+   choice between, manually installing Docker,
 
-```
-pip install pre-commit
-pre-commit install
-```
+   - on MacOS follow [these instructions](https://docs.docker.com/desktop/mac/install/)
+   - on Windows, [install WSL
+     2](https://docs.microsoft.com/en-us/windows/wsl/install), then Docker.
+     Note that Windows filesystem access from WSL2 is very slow and should
+     be avoided when building Pyodide.
+
+   The other alternative is to use a service that provides a Linux
+   development environment, such as,
```

Comment:
There should either be a colon or no punctuation here and in other places before display code.
```suggestion
   development environment, such as
```